undertow/src/main/java/org/wildfly/extension/undertow/UndertowRootDefinition.java

```diff
@@ -93,7 +96,13 @@ private UndertowRootDefinition() {
         super(UndertowExtension.SUBSYSTEM_PATH,
                 UndertowExtension.getResolver(),
                 UndertowSubsystemAdd.INSTANCE,
-                ReloadRequiredRemoveStepHandler.INSTANCE);
+                new ReloadRequiredRemoveStepHandler(){
+                    @Override
+                    protected void performRemove(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
+                        DefaultDeploymentMappingProvider.instance().clear();
+                        super.performRemove(context, operation, model);
+                    }
+                });
```

Comment:
I don't think this change should be made. This is the same as the host remove discussion we chatted about before I realized the host remove didn't require reload:

1) Deployment is deployed and is using various hosts.
2) User removes subsystem; we don't change runtime, we just report that reload is required.
3) User then triggers redeploy. But now there are no mappings so the deployment isn't available.

Now, in reality, removing the subsystem is going to recursively remove the hosts too, so they are going to go away, not requiring reload, so this doesn't really matter. But having this here can be a cause of future bugs, and clearing in subsystem add is enough to deal with the reload case, so I think that is sufficient.